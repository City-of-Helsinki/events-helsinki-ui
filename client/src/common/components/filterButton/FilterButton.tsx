import React from "react";
import { useTranslation } from "react-i18next";

import CloseIcon from "../../../icons/CloseIcon";
import styles from "./filterButton.module.scss";

export type FilterType = "category" | "publisher";

interface Props {
  onRemove: (value: string, type: FilterType) => void;
  text: string;
  type: FilterType;
  value: string;
}

const FilterButton: React.FC<Props> = ({ onRemove, text, type, value }) => {
  const { t } = useTranslation();

  const handleRemove = () => {
    onRemove(value, type);
  };

  return (
    <div className={styles.filter}>
      <button
        type="button"
        className={styles.closeButton}
        onClick={handleRemove}
        aria-label={t("commons.filter.ariaButtonRemove", {
          filter: text
        })}
      >
        <CloseIcon />
      </button>
      {text}
    </div>
  );
};

export default FilterButton;
