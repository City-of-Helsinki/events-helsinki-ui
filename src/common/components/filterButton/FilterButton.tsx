import classNames from 'classnames';
import { IconCross } from 'hds-react';
import React from 'react';
import { useTranslation } from 'react-i18next';

import styles from './filterButton.module.scss';

export type FilterType =
  | 'category'
  | 'date'
  | 'dateType'
  | 'division'
  | 'place'
  | 'publisher'
  | 'target'
  | 'text';

interface Props {
  onRemove: (value: string, type: FilterType) => void;
  text: string;
  type: FilterType;
  value: string;
}

const FilterButton: React.FC<Props> = ({ onRemove, text, type, value }) => {
  const { t } = useTranslation();

  const handleRemove = () => {
    onRemove(value, type);
  };

  return (
    <div className={classNames(styles.filter, styles[type])}>
      <button
        type="button"
        className={styles.closeButton}
        onClick={handleRemove}
        aria-label={t('commons.filter.ariaButtonRemove', {
          filter: text,
        })}
      >
        <IconCross size="s" aria-hidden />
      </button>
      {text}
    </div>
  );
};

export default FilterButton;
