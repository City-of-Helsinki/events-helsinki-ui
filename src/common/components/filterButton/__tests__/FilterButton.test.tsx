import { render } from '@testing-library/react';
import * as React from 'react';

import FilterButton from '../FilterButton';

test('matches snapshot', () => {
  const { container } = render(
    <FilterButton
      onRemove={jest.fn()}
      text="text"
      type="publisher"
      value="value"
    />
  );

  expect(container.firstChild).toMatchSnapshot();
});
