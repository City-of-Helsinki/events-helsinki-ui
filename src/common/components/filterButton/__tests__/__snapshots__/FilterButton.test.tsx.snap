// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="filter publisher"
>
  <button
    aria-label="Poista suodatin: text"
    class="closeButton"
    type="button"
  >
    <svg
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          d="M0 0h24v24H0z"
        />
        <path
          d="M17.657 7.757L13.414 12l4.243 4.243-1.414 1.414L12 13.414l-4.243 4.243-1.414-1.414L10.586 12 6.343 7.757l1.414-1.414L12 10.586l4.243-4.243z"
          fill="currentColor"
        />
      </g>
    </svg>
  </button>
  text
</div>
`;
